src/ImageRunModal.jsx

```diff
@@ -80,24 +80,30 @@ const PublishPort = ({ id, item, onChange, idx, removeitem, itemCount }) =>
                             </button>
                         </Popover>
                     }>
-                <TextInput id={id + "-host-port"}
-                            type='number'
-                            step={1}
-                            min={1}
-                            max={65535}
-                            value={item.hostPort || ''}
-                            onChange={value => onChange(idx, 'hostPort', value)} />
+                <NumberInput
+                        id={id + "-host-port"}
+                        min={1}
+                        max={65535}
+                        value={item.hostPort || ''}
+                        minusBtnAriaLabel={_("Decrease host port by one")}
+                        plusBtnAriaLabel={_("Increase host port by one")}
+                        onMinus={() => onMinusOne(idx, 'hostPort')}
+                        onPlus={() => onPlusOne(idx, 'hostPort')}
+                        onChange={ev => onChange(idx, 'hostPort', ev.target.value)} />
             </FormGroup>
             <FormGroup className="pf-m-3-col-on-md"
                         label={_("Container port")}
                         fieldId={id + "-container-port"} isRequired>
-                <TextInput id={id + "-container-port"}
-                            type='number'
-                            step={1}
-                            min={1}
-                            max={65535}
-                            value={item.containerPort || ''}
-                            onChange={value => onChange(idx, 'containerPort', value)} />
+                <NumberInput
+                        id={id + "-container-port"}
+                        min={1}
+                        max={65535}
+                        value={item.containerPort || ''}
```

Comment:
Numberinput's value should be a number

Reply:
Then we would need to default to 0 and I don't like that behavior. See in https://www.patternfly.org/v4/components/number-input Let's assume I want to enter there value `5`.

1. Select the current value
2. Delete it
3. It now has 0
4. Enter 5
5. It now has `05`. It is not possible to enter your own value without leading 0

Reply:
I know that it is defined as `number` but empty string just works :man_shrugging:

Reply:
As long as it doesn't give a console error it's fine I suppoes. But yeah there is no good default host/port for podman containers.

Reply:
it does not give any error in console